src/shared/Network/AsyncListener.hpp

```diff
@@ -31,7 +31,7 @@ namespace MaNGOS
     {
         public:
             // constructor for accepting connection from client
-            AsyncListener(boost::asio::io_service& io_service, std::string const& bindIp, unsigned short port) : m_service(io_service), m_acceptor(io_service, boost::asio::ip::tcp::endpoint(boost::asio::ip::address::from_string(bindIp), port))
+            AsyncListener(boost::asio::io_context& io_service, std::string const& bindIp, unsigned short port) : m_service(io_service), m_acceptor(io_service, boost::asio::ip::tcp::endpoint(boost::asio::ip::address::from_string(bindIp), port))
```

Comment:
The only (nitpick) comment I have is that maybe we should rename the parameter name as well, next to its type.

Reply:
What he said.